<?xml version='1.0'?>
<sdf version='1.6'>
  <world name='default'>
    <plugin name="ros_link_attacher_plugin" filename="libgazebo_ros_link_attacher.so"/>

    <light name='sun' type='directional'>
      <cast_shadows>1</cast_shadows>
      <pose frame=''>0 0 10 0 -0 0</pose>
      <diffuse>0.8 0.8 0.8 1</diffuse>
      <specular>0.2 0.2 0.2 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>-0.5 0.1 -0.9</direction>
    </light>
    
    <model name='ground_plane'>
      <static>1</static>
      <link name='link'>
        <collision name='collision'>
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>100 100</size>
            </plane>
          </geometry>
          <surface>
            <contact>
              <collide_bitmask>65535</collide_bitmask>
              <ode/>
            </contact>
            <friction>
              <ode>
                <mu>100</mu>
                <mu2>50</mu2>
              </ode>
              <torsional>
                <ode/>
              </torsional>
            </friction>
            <bounce/>
          </surface>
          <max_contacts>10</max_contacts>
        </collision>
        <visual name='visual'>
          <cast_shadows>0</cast_shadows>
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>5 5</size>
            </plane>
          </geometry>
          <material>
            <lighting>1</lighting>
            <shader type='vertex'>
              <normal_map>__default__</normal_map>
            </shader>
            <ambient>0.43 0.5 0.56 1</ambient>
            <diffuse>0.43 0.5 0.56 1</diffuse>
            <specular>0.01 0.01 0.01 1</specular>
            <emissive>0 0 0 1</emissive>
          </material>
        </visual>
        <self_collide>0</self_collide>
        <enable_wind>0</enable_wind>
        <kinematic>0</kinematic>
      </link>
    </model>
    
    <gravity>0 0 -9.8</gravity>
    <magnetic_field>6e-06 2.3e-05 -4.2e-05</magnetic_field>
    <atmosphere type='adiabatic'/>
    <physics name='default_physics' default='0' type='ode'>
      <max_step_size>0.001</max_step_size>
      <real_time_factor>1</real_time_factor>
      <real_time_update_rate>1000</real_time_update_rate>
    </physics>
    <scene>
      <ambient>0.4 0.4 0.4 1</ambient>
      <background>0.7 0.7 0.7 1</background>
      <shadows>1</shadows>
    </scene>
    <audio>
      <device>default</device>
    </audio>
    <wind/>
    <spherical_coordinates>
      <surface_model>EARTH_WGS84</surface_model>
      <latitude_deg>0</latitude_deg>
      <longitude_deg>0</longitude_deg>
      <elevation>0</elevation>
      <heading_deg>0</heading_deg>
    </spherical_coordinates>
    
    <!-- The table top of the robot -->
    <model name='table'>
      <link name='link'>
        <inertial>
          <mass>1</mass>
          <inertia>
            <ixx>0.166667</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>0.166667</iyy>
            <iyz>0</iyz>
            <izz>0.166667</izz>
          </inertia>
          <pose frame=''>0 0 0 0 -0 0</pose>
        </inertial>
        <pose frame=''>0 0 0 0 -0 0</pose>
        <gravity>1</gravity>
        <self_collide>0</self_collide>
        <kinematic>0</kinematic>
        <enable_wind>0</enable_wind>
        <visual name='visual'>
          <pose frame=''>0 0 0 0 -0 0</pose>
          <geometry>
            <box>
              <size>0.75 1.20 0.001</size>
            </box>
          </geometry>
          <material>
            <lighting>1</lighting>
            <shader type='vertex'>
              <normal_map>__default__</normal_map>
            </shader>
            <ambient>0.12 0.56 1 1</ambient>
            <diffuse>0.12 0.56 1 1</diffuse>
            <specular>0.01 0.01 0.01 1</specular>
            <emissive>0 0 0 1</emissive>
          </material>
          <transparency>0</transparency>
          <cast_shadows>1</cast_shadows>
        </visual>
        <collision name='collision'>
          <pose frame=''>0 0 0 0 -0 0</pose>
          <geometry>
            <box>
              <size>0.75 1.20 0.001</size>
            </box>
          </geometry>
        </collision>
      </link>
      <static>1</static>
      <allow_auto_disable>1</allow_auto_disable>
      <pose frame=''>0 0.525 0.0005 0 -0 0</pose>
    </model>

     <!-- Chess Board -->
    <model name="kaspy">
      <pose>0 -0.375 0.375  -1.57 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/gkasp.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>


    <!-- Chess Board -->
    <model name="chess_board">
      <pose>0 0.75 0  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/chess_board.dae</uri>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>

    <!-- Black Pieces -->
    <model name="rook_black_1">
      <pose>0.21 0.54 0.045  0 0 0</pose>
      <static>1</static>
      <link name='link'>
        <inertial>
          <mass>2</mass>
          <inertia>
            <ixx>0.166667</ixx>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyy>0.166667</iyy>
            <iyz>0</iyz>
            <izz>0.166667</izz>
          </inertia>
        </inertial>
        <gravity>1</gravity>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/rook_black.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>

    <model name="knight_black_1">
      <pose>0.15 0.54 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/knight_black.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>

    <model name="bishop_black_1">
      <pose>0.09 0.54 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/bishop_black.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="queen_black">
      <pose>0.03 0.54 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/queen_black.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="king_black">
      <pose>-0.03 0.54 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/king_black.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="bishop_black_2">
      <pose>-0.09 0.54 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/bishop_black.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="knight_black_2">
      <pose>-0.15 0.54 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/knight_black.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="rook_black_2">
      <pose>-0.21 0.54 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/rook_black.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>

     <model name="pawn_black_1">
      <pose>0.21 0.6 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/pawn_black.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="pawn_black_2">
      <pose>0.15 0.6 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/pawn_black.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="pawn_black_3">
      <pose>0.09 0.6 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/pawn_black.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="pawn_black_4">
      <pose>0.03 0.6 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/pawn_black.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="pawn_black_5">
      <pose>-0.03 0.6 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/pawn_black.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="pawn_black_6">
      <pose>-0.09 0.6 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/pawn_black.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="pawn_black_7">
      <pose>-0.15 0.6 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/pawn_black.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="pawn_black_8">
      <pose>-0.21 0.6 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/pawn_black.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>

    <!-- White Pieces -->
    <model name="rook_white_1">
      <pose>0.21 0.96 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/rook_white.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="knight_white_1">
      <pose>0.15 0.96 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/knight_white.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="bishop_white_1">
      <pose>0.09 0.96 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/bishop_white.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="queen_white">
      <pose>0.03 0.96 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/queen_white.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="king_white">
      <pose>-0.03 0.96 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/king_white.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="bishop_white_2">
      <pose>-0.09 0.96 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/bishop_white.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="knight_white_2">
      <pose>-0.15 0.96 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/knight_white.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="rook_white_2">
      <pose>-0.21 0.96 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/rook_white.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>

     <model name="pawn_white_1">
      <pose>0.21 0.9 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/pawn_white.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="pawn_white_2">
      <pose>0.15 0.9 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/pawn_white.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="pawn_white_3">
      <pose>0.09 0.9 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/pawn_white.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="pawn_white_4">
      <pose>0.03 0.9 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/pawn_white.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="pawn_white_5">
      <pose>-0.03 0.9 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/pawn_white.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="pawn_white_9">
      <pose>-0.09 0.9 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/pawn_white.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="pawn_white_7">
      <pose>-0.15 0.9 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/pawn_white.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    <model name="pawn_white_8">
      <pose>-0.21 0.9 0.045  0 0 0</pose>
      <static>true</static>
      <link name='link'>
        <visual name="visual">
          <geometry>
            <mesh><uri>file:///home/parallels/major_project/src/assignment_1/models/chess_board/meshes/pawn_white.dae</uri>
            <scale>0.5 0.5 0.5 </scale>
            </mesh>
          </geometry>
        </visual>
      </link>
    </model>
    
    <light type="point" name="point_light">
      <pose>0 0.75 1 0 0 0</pose>
      <diffuse>1 1 1 1</diffuse>
      <specular>.1 .1 .1 1</specular>
      <attenuation>
        <range>20</range>
        <linear>0.2</linear>
        <constant>0.15</constant>
        <quadratic>0.02</quadratic>
      </attenuation>
      <cast_shadows>false</cast_shadows>
    </light>
     <light type="spot" name="spot_light">
      <pose>0 0.2 1.37 -0.5 0 0</pose>
      <diffuse>0.5 0.5 0.5 1</diffuse>
      <specular>.1 .1 .1 1</specular>
      <attenuation>
        <range>20</range>
        <linear>0.2</linear>
        <constant>0.15</constant>
        <quadratic>0.02</quadratic>
      </attenuation>
      <spot>
        <inner_angle>0.6</inner_angle>
        <outer_angle>1</outer_angle>
        <falloff>1</falloff>
      </spot>
      <cast_shadows>false</cast_shadows>
    </light>
    

  </world>
</sdf>
 
